### PR TITLE
Improvement/browser refactor

### PR DESCRIPTION
**Pull request by andrepimenta**
**Description**

Browser refactoring and update of the native webview.

**Checklist**

* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍

**Review by rickycodes — APPROVED** [review]
tested this out locally. seems to function well!

ones less forked dep! :tada:

File: app/components/Views/BrowserTab/index.js [hunk 1]
```diff
@@ -1665,323 +1365,476 @@ export class BrowserTab extends PureComponent {
...
-					onCancel={this.onCancelWatchAsset}
-					onConfirm={this.onCancelWatchAsset}
-					suggestedAssetMeta={suggestedAssetMeta}
-				/>
-			</Modal>
-		);
+	/**
+	 * Add bookmark
+	 */
+	const addBookmark = () => {
```
Comment (estebanmino) [inline app/components/Views/BrowserTab/index.js:1419]: this could be missing the bookmarks passes as script

Comment (estebanmino) [reply app/components/Views/BrowserTab/index.js:1419]: image
cc @ibrahimtaveras00

Comment (estebanmino) [reply app/components/Views/BrowserTab/index.js:1419]: then on second load 

image

File: app/components/Views/BrowserTab/index.js [hunk 2]
```diff
@@ -487,25 +407,17 @@ export class BrowserTab extends PureComponent {
 			const rpcMethods = {
 				eth_requestAccounts: async () => {
```
Comment (estebanmino) [inline app/components/Views/BrowserTab/index.js:408]: this method is not working when calling it directly. for ex uniswap.eth 

image

File: app/components/Views/BrowserTab/index.js (hunk 1 above)
Comment (estebanmino) [reply app/components/Views/BrowserTab/index.js:1419]: disregard this comment

File: app/components/Views/BrowserTab/index.js [hunk 3]
```diff
@@ -1665,323 +1365,476 @@ export class BrowserTab extends PureComponent {
...
-}
+			{renderProgressBar()}
+			{!isHidden() && renderPhishingModal()}
+			{!isHidden() && renderUrlModal()}
+			{!isHidden() && renderApprovalModal()}
+			{!isHidden() && renderPhishingModal()}
+			{!isHidden() && renderWatchAssetModal()}
+			{!isHidden() && renderOptions()}
+			{!isHidden() && renderBottomBar()}
+			{!isHidden() && renderOnboardingWizard()}
```
Comment (estebanmino) [inline app/components/Views/BrowserTab/index.js:1714]: feels like we're calling this method too many times on `render`?

File: app/components/Views/BrowserTab/index.js (hunk 2 above)
Comment (estebanmino) [reply app/components/Views/BrowserTab/index.js:408]: same happend on prod

File: app/components/Views/BrowserTab/index.js [hunk 4]
```diff
@@ -1665,323 +1365,476 @@ export class BrowserTab extends PureComponent {
...
+				)}
 			</View>
-		);
-	}
-}
+			{renderProgressBar()}
+			{!isHidden() && renderPhishingModal()}
+			{!isHidden() && renderUrlModal()}
+			{!isHidden() && renderApprovalModal()}
+			{!isHidden() && renderPhishingModal()}
```
Comment (estebanmino) [inline app/components/Views/BrowserTab/index.js:1710]: two calls of `renderPhishingModal`

**Pull request merged**